$NetBSD: patch-CMakeLists.txt,v 1.3 2019/11/27 22:25:21 adam Exp $

Make so version match symlink versions.
Remove -Werror=conversion.
Do not build examples.

--- CMakeLists.txt.orig	2019-11-12 07:16:54.000000000 +0000
+++ CMakeLists.txt
@@ -74,7 +74,7 @@ project(JSONCPP
         LANGUAGES CXX)
 
 message(STATUS "JsonCpp Version: ${JSONCPP_VERSION_MAJOR}.${JSONCPP_VERSION_MINOR}.${JSONCPP_VERSION_PATCH}")
-set( JSONCPP_SOVERSION 22 )
+set( JSONCPP_SOVERSION ${JSONCPP_VERSION_MAJOR} )
 
 option(JSONCPP_WITH_TESTS "Compile and (for jsoncpp_check) run JsonCpp test executables" ON)
 option(JSONCPP_WITH_POST_BUILD_UNITTEST "Automatically run unit-tests as a post build step" ON)
@@ -141,9 +141,9 @@ endif()
 if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
     # using regular Clang or AppleClang
     if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.12.0)
-      add_compile_options(-Wall -Wconversion -Wshadow -Werror=conversion -Werror=sign-compare)
+      add_compile_options(-Wall -Wconversion -Wshadow -Werror=sign-compare)
     else()
-      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wconversion -Wshadow -Werror=conversion -Werror=sign-compare")
+      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wconversion -Wshadow -Werror=sign-compare")
     endif()
 elseif(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
     # using GCC
@@ -227,5 +227,3 @@ add_subdirectory( src )
 #install the includes
 add_subdirectory( include )
 
-#install the example
-add_subdirectory( example )
